module/zfs/dsl_destroy.c

```diff
@@ -945,7 +950,8 @@ dsl_destroy_head(const char *name)
 		 * remove the objects from open context so that the txg sync
 		 * is not too long.
 		 */
-		error = dmu_objset_own(name, DMU_OST_ANY, B_FALSE, FTAG, &os);
+		error = dmu_objset_own(name, DMU_OST_ANY, B_FALSE, B_TRUE,
```

Comment:
I remember fixing this a while ago.... Not sure what happened to that. I'll put it back to `B_FALSE`